spices/SPICE-0001-import-graph-analyzer-api.adoc

```diff
@@ -0,0 +1,371 @@
+
+Underneath the hood, the command builds a synthetic module, that ultimately uses standard library module <<stdlib-module-pkl-analyze,pkl.analyze>> to produce output.
+
+==== `--format`, `-f`
+
+The `--format` (`-f` for short) flag sets the output format for this command.
+
+Examples:
+
+[source,shellscript]
+----
+# Evaluate as JSON
+$ pkl analyze imports -f json myFile.pkl
+
+# Evaluate as YAML
+$ pkl analyze imports -f yaml myFile.pkl
+----
+
+==== Common CLI flags
+
+This command recursively traverses the import graph.
+To support this, the set of flags from common evaluator settings are available:
+
+* `--allowed-modules`
+* `--allowed-resources`
+* `--cache-dir`
+* `--module-path`
+* `--project-dir`
+* `--ca-certificates`
+
+[[stdlib-module-pkl-analyze]]
+=== Standard library module `pkl:analyze`
+
+A new standard library module is introduced.
+It contains the class `ImportAnalyzer`, which provides import graph analysis.
+
+[source,pkl]
+----
+/// Utilities for static analysis of Pkl modules.
+///
+/// Unlike `pkl:reflect`, these methods do not require loading the modules for analysis.
+@ModuleInfo { minPklVersion = "0.27.0" }
+@Since { version = "0.27.0" }
+module pkl.analyze
+
+/// Given a set of Pkl module URIs, returns a graph of imports declared by these modules.
+///
+/// The resulting graph includes transitive imports.
+external function importGraph(moduleUris: Set<Uri>): ImportGraph
+
+/// The graph of imports declared (directly and transitively) by the modules passed to
+/// [importGraph()].
+class ImportGraph {
+  /// The imports declared within a Pkl program.
+  ///
+  /// Each entry maps a module URI to the set of imports declared in that module.
+  ///
+  /// The set of all modules in the graph can be obtained via its [keys][Map.keys].
+  imports: Map<Uri, Set<Import>>
+
+  /// Mappings of modules from their in-language URI, to their resolved URI.
+  ///
+  /// A module's in-language URI is the form used within Pkl source code.
+  /// For example, modulepath-based modules have form `modulepath:/path/to/my/module.pkl`.
+  ///
+  /// A module's resolved URI is the form used to load the module's contents.
+  /// The same modulepath module might have form
+  /// `jar:file:///path/to/file.zip!/path/to/my/module.pkl` if Pkl run with
+  /// `--module-path /path/to/file.zip`.
+  ///
+  /// Dependency-notation imports, such as `"@myPackage/myModule.pkl"`, are represented as
+  /// in-language URIs with scheme `projectpackage:`.
+  /// In the case of local project dependenecies, they will be local URIs resolved from the project
+  /// file URI (in normal cases, `file:` URIs).
+  resolvedImports: Map<Uri, Uri>(keys == imports.keys)
+}
+
+/// An import as declared inside a module.
+class Import {
+  /// The absolute (in-language) URI of the import.
+  ///
+  /// Dependency notation URIs (such as `import "@foo/bar"`) are resolved to package URIs with
+  /// scheme `projectpackage:`.
+  uri: Uri
+}
```

Comment:
FYI: changed this design slightly, and introduced class `Import`.

This allows us to evolve this over time (perhaps add `sourceLocation`, `text`).